### PR TITLE
Header Component

### LINKED ISSUE #29: Header Component
## Common Component - Header :
The user is going to have a header on every page and the header contains a navigation bar to channel the page, Also the user profile icon, cart icon, login button, and the website logo.
- props: color, position.

- [ ]  It will be used in Pages :
almost every page.

- [ ]  Material UI link :
[Header link](https://material-ui.com/components/app-bar/)

- [ ] UI Image From Design :

image

### PR DESCRIPTION
**Pull request by AdhamHaisami**
Fix #29

# React Component

 - Create header component
 -  Create style file

### CODE REVIEW

File: client/.gitignore [hunk 1]
```diff
@@ -21,3 +21,5 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+.env
```
Comment (mFlaifel) [inline client/.gitignore:25]: insert empty line

File: client/package.json [hunk 2]
```diff
@@ -10,6 +10,7 @@
     "@testing-library/react": "^11.2.5",
     "@testing-library/user-event": "^12.8.3",
     "axios": "^0.21.1",
+    "eslint": "^7.23.0",
```
Comment (mFlaifel) [inline client/package.json:13]: why?

File: client/src/components/header/style.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  },
+  header: {
+    flexGrow: 1,
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-around',
+    alignItems: 'center',
+  },
+  title: {
+    // flexGrow: 1,
```
Comment (mFlaifel) [inline client/src/components/header/style.js:18]: remove commented code

**Review by mFlaifel — COMMENTED** [review]
- reduce header height
- activate buttons functionality
- add props that indicate if the user is logged or not.
- if you can increase the width of the component

**Review by imansedky5 — DISMISSED** [review]

**Review by khamis-hesham — APPROVED** [review]

**Review by yasmeenattallah — COMMENTED** [review]
- profile and cart icons should be added after login should you replace them with the signup navigation button? 
- try to use the logo in footer instead of Beauty word in center.
- check header  navigator

File: client/src/component/header/index.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                    </MenuItem>
+                    <MenuItem onClick={() => handleMenuClick('/blogs')}>
+                      Blogs
+                    </MenuItem>
+                    <MenuItem onClick={() => handleMenuClick('/contact')}>
+                      Contact
+                    </MenuItem>
+                    <MenuItem onClick={() => handleMenuClick('/cart')}>
+                      Cart
+                    </MenuItem>
```
Comment (mFlaifel) [inline client/src/component/header/index.js:83]: please remove anything related to cart page

File: client/src/component/header/index.js [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+  const handleMenu = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleMenuClick = (pageURL) => {
+    history.push(pageURL);
+    setAnchorEl(null);
+  };
+  const logOutClick = () => {};
```
Comment (mFlaifel) [inline client/src/component/header/index.js:34]: if you have time write logout function to logout the user and redirect him to home page

**Review by imansedky5 — APPROVED** [review]

**Review by mFlaifel — APPROVED** [review]

**Review by yasmeenattallah — APPROVED** [review]

**Review by khamis-hesham — APPROVED** [review]

**Pull request merged**